### DISCUSSION TITLE
Syncing Multiple Requests

### DISCUSSION THREAD

**Discussion by 1cg** [bigskysoftware/htmx#687] [Q&A] [answered]
Hey everyone 👋 , I'm wondering about how htmx solves a specific race condition that I was previously avoiding using plain JS. It happens when one of our form elements triggers a request on change. The server responds with HTML, which is then replaced on the page.

The race condition happens in this scenario:
- User selects checkbox 1, request 1 goes out
- User in addition selects checkbox 2, request 2 goes out
- Request 2 comes back first, the HTML is replaced. The HTML response has details about checkbox 1 and checkbox 2
- Request 1 comes back, with HTML that only contains details about checkbox 1.

The delay  modifier doesn't solve this issue in theory because even if there is a delay of say 500ms between checkbox 1 and 2 being selected, it's still technically possible for request 2 to come back first.

The way we were solving this via plain JS was to store the request time of the most recent request and ignore any updates that were < that request time.

Does htmx offer anything like that solution?

**Comment by 1cg** [comment, marked as answer]
The way to fix that is to hoist the request up the DOM to a parent element and then use a [`from: <CSS selector>`](https://htmx.org/attributes/hx-trigger/) that matches both checkboxes.  

This will use a single request queue for all requests based on the parent element.

**Reply by shawnwall** [reply]
can someone provide an example here please? I have a separate issue raised [here](https://github.com/bigskysoftware/htmx/discussions/2834) where I couldn't figure out a solution.
